scripts/translation/genrevdb.php

```diff
@@ -0,0 +1,252 @@
+ *  | Copyright (c) 1997-2023 The PHP Group                                |
+ *  +----------------------------------------------------------------------+
+ *  | This source file is subject to version 3.01 of the PHP license,      |
+ *  | that is bundled with this package in the file LICENSE, and is        |
+ *  | available through the world-wide-web at the following url:           |
+ *  | https://www.php.net/license/3_01.txt.                                |
+ *  | If you did not receive a copy of the PHP license and are unable to   |
+ *  | obtain it through the world-wide-web, please send a note to          |
+ *  | license@php.net, so we can mail you a copy immediately.              |
+ *  +----------------------------------------------------------------------+
+ *  | Authors:     André L F S Bacci <ae php.net>                          |
+ *  +----------------------------------------------------------------------+
+ *  | Description: Check format for revtags and credits on XML comments.   |
+ *  +----------------------------------------------------------------------+
+ */
+
+require_once __DIR__ . '/lib/all.php';
+
+if ( count( $argv ) < 3 || in_array( '--help' , $argv ) || in_array( '-h' , $argv ) )
+{
+    fwrite( STDERR , "\nUsage: {$argv[0]} [file.db] [lang1,langN]\n\n" );
+    return;
+}
+
+$timeStart = new \DateTime;
+$dbpath = $argv[1];
+$langs = array();
+for( $idx = 2 ; $idx < count( $argv ) ; $idx++ )
+    $langs[] = $argv[ $idx ];
+
+consolelog( "Creating revdata database $dbpath for languages: " . implode( ',', $langs ) . '.');
+
+$db = db_create( $dbpath );
+foreach( $langs as $lang )
+    generate( $db , $lang );
+
+consolelog( "Revdata database $dbpath complete." );
+exit;
+
+
+
+function generate( SQLite3 $db , string $lang )
+{
+    $cwd = getcwd();
+    if ( ! is_dir( $lang )  )
+    {
+        consolelog( "Error: '$cwd/$lang' doesn't exist. Skipped." );
+        return;
+    }
+    if ( ! is_file( "$lang/translation.xml" ) )
+    {
+        consolelog( "Error: '$cwd/$lang' doesn't contains translation.xml. Skipped." );
+        return;
+    }
+
+    try
+    {
+        consolelog( "Language $lang started." );
+
+        $revcheck = new RevcheckRun( 'en' , $lang );
+        $data = $revcheck->revData;
+
+        $db->exec( 'BEGIN TRANSACTION' );
+
+        db_insert( $db , "languages" , $data->lang , $data->intro );
+
+        foreach( $data->translators as $translator )
+            if ( $translator->nick != "" )
+                db_insert( $db , "translators", $data->lang
+                    , $translator->name
+                    , $translator->nick
+                    , $translator->email
+                    , $translator->vcs
+                    , $translator->countOk
+                    , $translator->countOld
+                    , $translator->countOther
+                );
+
+        foreach( $data->fileDetail as $file )
+            if ( $translator->nick != "" )
```

Comment:
This test shouldn't be here, it's not relevant for this loop

Reply:
Good catch, fixed.